src/mapper.rs

```diff
@@ -0,0 +1,224 @@
+use std::collections::{BTreeMap, HashMap};
+use std::{fmt::Write, iter::FusedIterator};
+
+use crate::mapping::{MappingRecord, ProguardMapping};
+use crate::stacktrace::StackFrame;
+
+#[derive(Debug)]
+struct MemberMapping<'s> {
+    startline: usize,
+    endline: usize,
+    original_class: Option<&'s str>,
+    original: &'s str,
+    original_startline: usize,
+    original_endline: Option<usize>,
+}
+
+#[derive(Debug)]
+struct ClassMapping<'s> {
+    original: &'s str,
+    obfuscated: &'s str,
+    members: BTreeMap<&'s str, Vec<MemberMapping<'s>>>,
+}
+
+type MemberIter<'m> = std::iter::Fuse<std::slice::Iter<'m, MemberMapping<'m>>>;
+
+/// An Iterator over remapped StackFrames.
+#[derive(Debug)]
+pub struct RemappedFrameIter<'m> {
+    inner: Option<(StackFrame<'m>, MemberIter<'m>)>,
+}
+
+impl<'m> RemappedFrameIter<'m> {
+    fn empty() -> Self {
+        Self { inner: None }
```

Comment:
_nit_: Also implement `Default` that uses `Self::empty()` internally. This can be extremely useful when using this Iterator in higher-level iterators.